transactions/src/main/java/org/ehcache/transactions/xa/internal/xml/TxCacheManagerServiceConfigurationParser.java

```diff
@@ -71,11 +83,42 @@ public URI getNamespace() {
 
   @Override
   public Class<TransactionManagerProvider> getServiceType() {
-    return null;
+    return TransactionManagerProvider.class;
   }
 
   @Override
   public Element unparseServiceCreationConfiguration(ServiceCreationConfiguration<TransactionManagerProvider> serviceCreationConfiguration) {
-    throw new UnsupportedOperationException("Not yet implemented");
+    try {
+      validateParameterForTranslationToServiceConfig(serviceCreationConfiguration);
+      Document doc = createDocumentRoot();
+      LookupTransactionManagerProviderConfiguration lookupTransactionManagerProviderConfiguration = (LookupTransactionManagerProviderConfiguration) serviceCreationConfiguration;
+      Element rootElement = createRootElement(doc, lookupTransactionManagerProviderConfiguration);
+      return rootElement;
+    } catch (SAXException | ParserConfigurationException | IOException e) {
+      throw new XmlConfigurationException(e);
+    }
+  }
+
+  private void validateParameterForTranslationToServiceConfig(ServiceCreationConfiguration<TransactionManagerProvider> serviceCreationConfiguration) {
```

Comment:
This seems a little like overkill since almost the next thing you do it to cast the thing and then dereference it... if you'd like to keep this method however you could at least make it finish the job and return the cast instance so we're only type checking/casting in one place.